src/lightning/pytorch/trainer/connectors/data_connector.py

```diff
@@ -420,36 +419,17 @@ def _check_dataloader_iterable(
         )
 
 
-def _worker_check(dataloader: object, using_spawn: bool, name: str) -> None:
+def _worker_check(trainer: "pl.Trainer", dataloader: object, name: str) -> None:
     if not isinstance(dataloader, DataLoader):
         return
 
-    num_cpus = multiprocessing.cpu_count()
-
-    # ddp_spawn + num_workers > 0 don't mix! tell the user
-    if dataloader.num_workers > 0 and using_spawn:
-        if not dataloader.persistent_workers:
-            rank_zero_warn(
-                "num_workers>0, persistent_workers=False, and strategy=ddp_spawn"
-                " may result in data loading bottlenecks."
-                " Consider setting persistent_workers=True"
-                " (this is a limitation of Python .spawn() and PyTorch)"
-            )
-
-    elif dataloader.num_workers == 0 and using_spawn:
-        if not dataloader.persistent_workers:
-            rank_zero_warn(
-                "strategy=ddp_spawn and num_workers=0 may result in data loading bottlenecks."
-                " Consider setting num_workers>0 and persistent_workers=True"
-            )
```

Comment:
What's the reason behind removing this?

Reply:
These are legacy, from a time when 1) ddp-spawn was the default 2) spawn wasn't the first thing we did in fit and multiprocessing defaults in DataLoader were different. 

Furthermore, the persistent_workers setting is irrelevant. It has nothing to do with spawned processes. No reason was given in the original PR why this is needed: https://github.com/Lightning-AI/lightning/pull/6762/

I don't see issues any of these settings. A warning IMO is not appropriate and would only confuse users.

Reply:
It might be motivated by this thread: https://github.com/pytorch/pytorch/issues/15849#issuecomment-810876377 (author commented)
I found another discussions: https://github.com/pytorch/ignite/issues/1938.
It might be worth benchmarking before removing it

If you are confident about removing it, you will want to remove it from https://lightning.ai/docs/pytorch/stable/advanced/speed.html#persistent-workers too

Reply:
I am confident that this info is outdated in recent torch versions, but I'll do it in a separate PR where I demonstrate this with some code examples. Resolving